src/application/service/tests/UpdateTestType.test.ts

```diff
@@ -0,0 +1,80 @@
+import database from '../../../database';
+import { testTypeRepository } from '../../../infrastructure/persistence';
+import { cleanupDatabase } from '../../../test/cleanupDatabase';
+import { UpdateTestType } from './UpdateTestType';
+import { TestTypeId } from '../../../domain/model/test/testType/TestTypeId';
+import { TestTypeNotFoundError } from '../../../domain/model/test/testType/TestTypeRepository';
+import { v4 } from 'uuid';
+import { aTestType } from '../../../test/domainFactories';
+import { InterpretationRules } from '../../../domain/model/test/interpretation/InterpretationRules';
+
+describe('UpdateTestType', () => {
+  const updateTestType = new UpdateTestType(testTypeRepository);
+
+  beforeEach(async () => {
+    await cleanupDatabase();
+  });
+
+  it('throws not found error if test type id does not exist', async () => {
+    const testTypeId = new TestTypeId().value;
+    await expect(updateTestType.execute(testTypeId, { name: 'NEW_NAME' })).rejects.toEqual(
+      new TestTypeNotFoundError(testTypeId)
+    );
+  });
+
+  it('updates the test type name', async () => {
+    const testType = await testTypeRepository.save(aTestType());
+    const newName = v4();
```

Comment:
Little bit weird?  Introduces non-determinism into the test, I know it doesn't (shouldn't...) matter in this case, but maybe just simpler all round to hard code a string?

Reply:
Ha! I knew someone would catch this :) Question: isn't there less non-determinism? As in, isn't this a bigger guarantee that the name will be a unique string vs a hard coded value (which might colide with whatever the `aTestType()` factory returns?